Project-Aurora/Project-Aurora/Devices/YeeLight/YeeLightDevice.cs

```diff
@@ -118,27 +116,43 @@ private int GetFreeTCPPort()
 
         public void Connect(IPAddress lightIP, DoWorkEventArgs token = null)
         {
-            try
+            if (light.IsConnected())
             {
-                if (!light.isConnected())
-                {
-                    IPAddress localIP;
-                    int localListenPort = GetFreeTCPPort(); // This can be any port
-
-                    using (Socket socket = new Socket(AddressFamily.InterNetwork, SocketType.Dgram, 0))
-                    {
-                        socket.Connect(lightIP, lightListenPort);
-                        localIP = ((IPEndPoint)socket.LocalEndPoint).Address;
-                    }
+                return;
+            }
 
-                    isConnected = light.Connect(lightIP, lightListenPort) && light.SetMusicMode(localIP, localListenPort);
+            //Auto discover a device if the IP is 0.0.0.0
+            if (lightIP.Equals("0.0.0.0"))
+            {
+                List<YeeLightAPI.YeeLightDevice> devices = YeeLightAPI.YeeLightDeviceLocator.DeviceLocator.DiscoverDevices();
+                if (devices.Any())
+                {
+                    light = devices.First();
                 }
+                else
+                {
+                    throw new Exception("IP address is set to 0.0.0.0 and no devices has been located.");
+                }
+            }
+            //If it isn't then set the IP to the provided IP address
+            else
+            {
+                light.SetLightIPAddressAndPort(lightIP, YeeLightAPI.YeeLightConstants.Constants.DefaultCommandPort);
             }
-            catch (Exception exc)
+
+            IPAddress localIP;
+            int localListenPort = GetFreeTCPPort(); // This can be any free port
+
+            using (Socket socket = new Socket(AddressFamily.InterNetwork, SocketType.Dgram, 0))
             {
-                Global.logger.Error($"Device {devicename} encountered an error during Connecting. Exception: {exc}");
-                isConnected = false;
+                socket.Connect(lightIP, lightListenPort);
```

Comment:
If the IP is 0.0.0.0, so discovery feature is on,  the lightIP is not the discovered device IP. This case it's always 0.0.0.0, I am not sure it is going to cause any problem.

Reply:
Thanks, good finding, I'll try to fix this, maybe even that's the reason why device discover wasn't working correctly.